### ISSUE TITLE
More organized/robust parsing with Alex and Happy - Preliminary Work

### ISSUE THREAD

**Issue by ChrisEPhifer** [edwisdom/chords#6]
`Parsec` is more than acceptable for this stage of the project; it's intuitive, and only rarely do we rely on tricky things like the `try` combinator. 

As you think about what you want to be able to express in the language other than chords, though, the task of building correct parsers with combinators alone gets a bit out of hand. This is where `alex` and `happy`, descendants of the `lex` and `yacc` lexer and parser generators, come in.

Here's what would help a lot with that eventual transition:

- [ ] Identify the language's tokens
- [ ] Write a BNF grammar for the language (this is likely a long-term WIP)

**Comment by ChrisEPhifer** [comment]
@edwisdom I'll take this ticket once #17 is merged into `develop` since it has some updates to the current parser. Shouldn't take terribly long, I've had to work with these tools at work and they're not the complete worst things in the world :)
